src/operator/tensor/matrix_op.cc

```diff
@@ -23,57 +23,51 @@ DMLC_REGISTER_PARAMETER(ReverseParam);
 
 NNVM_REGISTER_OP(Reshape)
 .add_alias("reshape")
-.describe(R"code(Reshape array into a new shape.
+.describe(R"code(Reshapes Array into a new shape.
 
-The shape is a tuple of int such as (2,3,4). The new shape should not change the
-array size. For example::
+Given an array and a shape, this function returns a copy of the array in the new shape.
 
-   reshape([1,2,3,4], shape=(2,2)) = [[1,2], [3,4]]
-
-In addition, we can use special codes, which are integers less than
-1, on some shape dimensions. To inference the output shape, we set it to an
-empty tuple at beginning. When continuously pop dimensions from the original
-shape starting from the beginning, and then push translated results into the output
-shape.
-
-Each special code presents a way of translation.
-
-- ``0`` for copying one. Pop one input dimension and push into the output. For example::
-
-  - input=(2,3,4), shape=(4,0,2), output=(4,3,2)
-  - input=(2,3,4), shape=(2,0,0), output=(2,3,4)
-
-- ``-1`` for inference. Push a placeholder into the output whose value will be inferred later::
-
-  - input=(2,3,4), shape=(6,1,-1), output=(6,1,4)
-  - input=(2,3,4), shape=(3,-1,8), output=(3,1,8)
-  - input=(2,3,4), shape=(-1,), output=(24,)
+The shape is a tuple of integers such as (2,3,4). The size of the new shape should be same as the size of the input array.
 
-- ``-2`` for copying all. Pop all remaining input dimensions and push them into
-  the output::
-
-  - input=(2,3,4), shape=(-2), output=(9,8,7)
-  - input=(2,3,4), shape=(2,-2), output=(2,3,4)
-  - input=(2,3,4), shape=(-2,1,1), output=(2,3,4,1,1)
-
-- ``-3`` for merging two dimensions. Pop two input dimensions, compute the product and then
-  push into the output::
-
-  - input=(2,3,4), shape=(-3,4), output=(6,4)
-  - input=(2,3,4), shape=(0,-3), output=(2,12)
-  - input=(2,3,4), shape=(-3,-2), output=(6,4)
-
-- ``-4`` for splitting two dimensions. Pop one input dimensions, next split it
-  according to the next two dimensions (can contain one ``-1``) specified after
-  this code, then push into the output::
-
-  - input=(2,3,4), shape=(-4,1,2,-2), output=(1,2,3,4)
-  - input=(2,3,4), shape=(2,-4,-1,3,-2), output=(2,1,3,4)
+Example::
+   reshape([1,2,3,4], shape=(2,2)) = [[1,2], [3,4]]
 
-If the argument ``reverse`` is set to be true, then translating the input shape
-from right to left. For example, with input shape (10, 5, 4) target shape (-1,
-0), then the output shape will be (50,4) if ``reverse=1``, otherwise it will be
-(40,5).
+Some dimensions of the shape can have special values of 0, -1, -2, -3 or -4 which is interpreted as below: 
+
+- ``0`` copy one dimension from the input to the output shape.
```

Comment:
"copy one dimension ..."  -> "copy this dimension from the input shape to the output shape"